tests/conftest.py

```diff
@@ -455,3 +456,52 @@ def disable_container_autorestart(duthost, request):
     for name, state in container_autorestart_states.items():
         if state == "enabled":
             duthost.command(cmd_enable.format(name))
+
+def generate_param_asic_index(request, dut_indices, param_type):
+    logging.info("generating {} asic indicies for  DUT [{}] in ".format(param_type, dut_indices))
+    inv_file =  request.config.getoption("ansible_inventory")
+    tbname = request.config.getoption("--testbed")
+    tbfile = request.config.getoption("--testbed_file")
+    if tbname is None or tbfile is None:
+        raise ValueError("testbed and testbed_file are required!")
+    
+    with open(inv_file, 'r') as fh:
+        inv = yaml.safe_load(fh)
+
+    hosts = inv['all']['children']['sonic']['hosts']
+    tbinfo = TestbedInfo(tbfile)
+
+    #if the params are not present treat the device as a single asic device
+    asic_index_params = [DEFAULT_ASIC_ID]
+
+    for dut_id in dut_indices:
+        dut = tbinfo.testbed_topo[tbname]["duts"][dut_id]
+        inv_data = hosts[dut]
+        if param_type == ASIC_PARAM_TYPE_ALL and ASIC_PARAM_TYPE_ALL in inv_data:
+            asic_index_params = range(int(inv_data[ASIC_PARAM_TYPE_ALL]))
+        if param_type == ASIC_PARAM_TYPE_FRONTEND and ASIC_PARAM_TYPE_ALL in inv_data:
+            asic_index_params = inv_data[ASIC_PARAM_TYPE_ALL]
```

Comment:
It would be more robust to use ansible.inventory.manager.InventoryManager to load inventory file and get host variables. Example: https://github.com/Azure/sonic-mgmt/blob/master/ansible/library/testbed_vm_info.py#L91
Using yaml.safe_load assumes that the inventory file must be yaml format. But anyway it is OK to use yaml.safe_load for now.

Reply:
updated to use the inventory manager.